src/Microsoft.Health.Fhir.Core/Features/Operations/Reindex/ReindexJobTask.cs

```diff
@@ -463,8 +485,13 @@ private async Task CalculateTotalCount(CancellationToken cancellationToken)
                 SearchResult countOnlyResults = await ExecuteReindexQueryAsync(queryForCount, countOnly: true, cancellationToken);
                 if (countOnlyResults != null)
                 {
+                    _reindexJobRecord.ResourceCounts.TryAdd(resourceType, countOnlyResults.TotalCount.Value); // TODO: What action should be taken if this fails?
```

Comment:
Since this is a performance improvement, if it happens to fail, we can simply go ahead with the query for this resource type.  But we will need to protect line 156 above, in case this resource type doesn't make it into the dictionary.

Reply:
Makes sense, thanks. Fixed in a8a26fb.